### PR TITLE
chore: Simplify env files.

### PR DESCRIPTION
**Pull request by cullylarson**
## Changes

- fix: A few `copyFile` entries mistakenly target the wrong files.
- chore: Remove `.env.local`. We shouldn't ever need `.env.local` because it will be applied at every stage (not usually intended).
- chore: Don't override values in `env.*.local` unless needed.
- chore: Don't set any values in `.env`. Use it as basically a template. The value in `.env` will be applied to every environment, which doesn't usually make sense.

## Checklist

- [ ] Requires dependency update?
- [x] Generating a new app works

### CODE REVIEW

**Review by alexanderson1993 — APPROVED** [review]

File: packages/create-bison-app/template/_.env.test.ejs [hunk 1]
```diff
@@ -1,10 +1,8 @@
-# Copy to .env.test.local to override
-# Include NODE_ENV for package script withEnv to work as expected
+# Override values in .env.development.local
```
Comment (cullylarson) [inline packages/create-bison-app/template/_.env.test.ejs]: Fixed this. Will push with any other changes.

File: packages/create-bison-app/template/_.env.ejs [hunk 2]
```diff
@@ -1,19 +1,8 @@
-# Copy to .env.local to mimic production
+APP_ENV=
```
Comment (kgajera) [inline packages/create-bison-app/template/_.env.ejs]:
```suggestion
```

File: packages/create-bison-app/template/_.env.development.ejs [hunk 3]
```diff
@@ -1,17 +1,8 @@
...
-# Environment variables declared in this file are automatically made available to Prisma.
-# See the documentation for more detail: https://pris.ly/d/prisma-schema#using-environment-variables
-
-# Prisma supports the native connection string format for PostgreSQL, MySQL and SQLite.
-# See the documentation for all the connection string options: https://pris.ly/d/connection-strings
-
-# Include NODE_ENV for package script withEnv to work as expected
+# Override values in .env.development.local
 
 APP_ENV="development"
```
Comment (kgajera) [inline packages/create-bison-app/template/_.env.development.ejs]:
```suggestion
```

**Review by kgajera — APPROVED** [review]
Unless I've missed something, we can remove `APP_ENV` after #312 is merged.

**Pull request merged**